### DISCUSSION TITLE
Problems with Degree Sign

### DISCUSSION THREAD

**Discussion by AirTobe91** [Bodmer/TFT_eSPI#2102] [Q&A - Fonts]
Hello,

I use the new Lilygo T-Display S3 with Platform.io in Visual Studio Code.
I'm using sprites to display angles in degree and used the "°" icon for the values, e.g. -> -0.3°

This works fine with the standart font in size 2 and 4.

```
#include "Arduino.h"
#include "TFT_eSPI.h"
#include "pin_config.h"

TFT_eSPI tft = TFT_eSPI();
TFT_eSprite testSprite = TFT_eSprite(&tft);

void setup() {
  tft.begin();
  tft.setRotation(3);
  tft.fillScreen(TFT_BLACK);
  testSprite.createSprite(320, 170);
  testSprite.fillSprite(TFT_BLACK);
  testSprite.drawCentreString("-0.3°", 160, 85, 4);
  testSprite.pushSprite(0,0);
}

//loops from 0 -> 5.0 -> -5.0 -> 0 and back
void loop() {

  for(float i=0.0f; i<5.0; i+=0.1f)
  {
    displayValue(i);
  }
  for (float i = 5.0f; i > -5.0; i -= 0.1f) 
  {
    displayValue(i);
  }
  for (float i = -5.0f; i < 0.0; i += 0.1f)
  {
    displayValue(i);
  }
}

void displayValue(float angle)
{
// limit float angle decimal values to 1 digit and convert to string again
  char angleChar[4];
  sprintf(angleChar, "%.1f", angle);
  String angleStr(angleChar);
// Due to it's not possible to use ° char directly in concatentation, use ` instead which displays a ° later on in display (based on a other issue raised here)
  angleStr += "`";
  testSprite.fillSprite(TFT_BLACK);
  testSprite.drawCentreString(angleStr, 160, 85, 4);
  testSprite.pushSprite(0, 0);
}

```

This code successfully displays the values form -5.0°- 5.0° in the display, for font size 2 and 4.
When I try to change font size to 6, the ° sign disappears, the rest still works fine.

Can somebody please help me with this problem?
Thank you,
Tobias!

**Comment by Bodmer** [comment]
The sign disappears because font 6 only contains a few characters to display numerical values and time, as noted here:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setup.h#L307-L309

Bear in mind these font files were restricted in characters as they were originally created for the Arduino UNO which has a severely limited FLASH size.

**Comment by Bodmer** [comment]
Forgot to mention that I have got around this issue before by using a small letter "o" from another font to be the degree symbol.

See for example [this image](https://camo.githubusercontent.com/b65a710352ad254a1a461ac23ec579477042cc0d1ab174ce1ae900cdbc0aa6a0/68747470733a2f2f692e696d6775722e636f6d2f4f526f76774e592e706e67) where °C is done that way.
